Artesian/_ClientsExecutor/RequestExecutor.py

```diff
@@ -251,4 +252,20 @@ def __init__(self, last_attempt):
         self.last_attempt = last_attempt
 
     def __str__(self):
-        return "RetryError[{0}]".format(self.last_attempt)    
\ No newline at end of file
+        return "RetryError[{0}]".format(self.last_attempt)    
+
+
+def createSemaphore(parallelism):
+    """
+    Wrapper around asyncio semaphore.
+    Ensures that there is an even loop available.
+    An event loop may not be available if the sdk is not run in the main event loop
```

Comment:
This is strange, the Event loop should always be present at the time of issuing a Request.

Could be that the solution to the problem would be to defer the creation of  Semaphore (or the _RequestExecutor) to later time in respect to now? For example to the first request?

The nonAsync Execute of the query is already supposed to do this. See https://github.com/ARKlab/Artesian.SDK-Python/blob/c4df9dfb331260a774e8669948e5bb5fdde8b26c/Artesian/_Query/Query.py#L55

Reply:
Delaying creation won't help. The same situation will occur at the time of the first request. If you think lazy evaluation of the semaphore makes sense to do anyway I can add it

Reply:
Why at the time of the first request would still happen? 

If is in an `async `chain from the user code, for sure the program is running with a loop as the user is calling `executeAsync()`
Otherwise the user is calling the Sync 'execute()' that is supposed to start the loop and wait the `_execAsync()`

By the time it reach the _RequestExecutor it should always have a loop.

What I mean is strange to me that is a 'createSemaphore', that is just a policy handler, to ensure that runtime requirements are sane.

I see a more correct solution (unless there is something I'm missing) to:
- use asyncio.run() in syncronouse execute() or the logic you made here (try get loop or create one)
- defer _RequestExecutor (or Semaphore) _after_ the call to execute() is made

This way the runtime is _always_ in a coroutine context and the `execute()` is the compatibility layer as was supposed to be from start.